### PR TITLE
Dynamic model

### PR DESCRIPTION
**Pull request by mediumTaj**
Added DynamicModel for arbitrary (unrestricted) AdditionalProperties

### CODE REVIEW

File: src/IBM.Cloud.SDK.Core/Model/DynamicModel.cs [hunk 1]
```diff
@@ -56,7 +59,50 @@ public void Remove(string key)
...
-            AdditionalProperties.TryGetValue(key, out T value);
+            AdditionalProperties.TryGetValue(key, out JToken value);
+            return value.ToObject<T>();
+        }
+    }
+
+    /// <summary>
+    /// This class is the base class for generated models with arbitrary additional properties.
+    /// </summary>
+    /// <typeparam name="T"></typeparam>
```
Comment (lpatino10) [inline src/IBM.Cloud.SDK.Core/Model/DynamicModel.cs]: This line isn't necessary for the unrestricted version.

File: src/IBM.Cloud.SDK.Core/Model/DynamicModel.cs [hunk 2]
```diff
@@ -56,7 +59,50 @@ public void Remove(string key)
...
+    public class DynamicModel
+    {
+        /// <summary>
+        /// A Dictionary to keep arbitrary additional properties.
+        /// </summary>
+        [JsonExtensionData]
+        public Dictionary<string, JToken> AdditionalProperties { get; } = new Dictionary<string, JToken>();
```
Comment (lpatino10) [inline src/IBM.Cloud.SDK.Core/Model/DynamicModel.cs:76]: Is this arbitrary version of `DynamicModel` being handled differently anywhere? It all seems to be the same code.

**Review by lpatino10 — CHANGES_REQUESTED** [review]
Just left a couple comments because I can't tell what's changing between the restricted and unrestricted versions of the class.

File: src/IBM.Cloud.SDK.Core/Model/DynamicModel.cs (hunk 1 above)
Comment (mediumTaj) [reply src/IBM.Cloud.SDK.Core/Model/DynamicModel.cs]: oops you're right - removed!

File: src/IBM.Cloud.SDK.Core/Model/DynamicModel.cs (hunk 2 above)
Comment (mediumTaj) [reply src/IBM.Cloud.SDK.Core/Model/DynamicModel.cs:76]: The only difference in the two classes are in `Add` and `Get` where the restricted to the designated type or a generic object. The rest of the class is the same. The `AdditionalProperties` prop would be restricted to type but in order to use `JsonExtensionData` we need this to be a `JToken`. Conversion to type is done in the `Get` method.

Comment (lpatino10) [reply src/IBM.Cloud.SDK.Core/Model/DynamicModel.cs:76]: Ohh okay I see now, sorry about that! Makes sense.

**Review by lpatino10 — APPROVED** [review]
Thanks for clearing up my confusion! Looks good.

**Pull request merged**